pkg/window/windower.go

```diff
@@ -0,0 +1,19 @@
+package window
+
+import (
+	"time"
+)
+
+// Windower assigns the element to zero or more windows.
+type Windower interface {
+	// AssignWindow assigns the event to the window based on give window configuration.
+	AssignWindow(eventTime time.Time) []*IntervalWindow
+}
+
+// IntervalWindow has the window boundary details.
+type IntervalWindow struct {
```

Comment:
The name `IntervalWindow` is confusing when it is together with `KeyedWindow`, see the code below.

```
func (aw *ActiveWindows) CreateKeyedWindow(iw *window.IntervalWindow) *keyed.KeyedWindow {
```

Reply:
i am going to leave this as is for now. Will tackle this when we go about implementing sliding windows. 
may be bounded and unbounded windows make more sense.